### PR TITLE
CM-1020 - Add EID translation for sovrn

### PR DESCRIPTION
**Pull request by 3link**
Related PRs:
- https://github.com/LiveIntent/idres/pull/1411
- https://github.com/LiveIntent/prebid.github.io/pull/20
- https://github.com/LiveIntent/castle/pull/449

### CODE REVIEW

**Review by wi101 — APPROVED** [review]

File: test/spec/modules/eids_spec.js [hunk 1]
```diff
@@ -238,6 +238,40 @@ describe('eids array generation for known sub-modules', function() {
...
+  });
+
+  it('sovrn with ext', function() {
+    const userId = {
+      sovrn: {'id': 'sample_id', 'ext': {'provider': 'some.provider.com'}}
+    };
+    const newEids = createEidsArray(userId);
+    expect(newEids.length).to.equal(1);
+    expect(newEids[0]).to.deep.equal({
+      source: 'media.net',
```
Comment (peixunzhang) [inline test/spec/modules/eids_spec.js:264]:
```suggestion
      source: 'liveintent.sovrn.com',
```

File: test/spec/modules/eids_spec.js [hunk 2]
```diff
@@ -238,6 +238,40 @@ describe('eids array generation for known sub-modules', function() {
...
+      uids: [{
+        id: 'sample_id',
+        atype: 3
+      }]
+    });
+  });
+
+  it('sovrn with ext', function() {
+    const userId = {
+      sovrn: {'id': 'sample_id', 'ext': {'provider': 'some.provider.com'}}
```
Comment (peixunzhang) [inline test/spec/modules/eids_spec.js:259]:
```suggestion
      sovrn: {'id': 'sample_id', 'ext': {'provider': 'liveintent.sovrn.com'}}
```

File: test/spec/modules/eids_spec.js (hunk 1 above)
Comment (3link) [reply test/spec/modules/eids_spec.js]: Good catch. Some copy pasta. But the test run was green. Hmm. 

Fixed this one and the one in the ext below in https://github.com/LiveIntent/Prebid.js/pull/26/commits/664636e26c9700c853d2ff14b507dc506202dfce.

File: test/spec/modules/eids_spec.js (hunk 2 above)
Comment (3link) [reply test/spec/modules/eids_spec.js:258]: That should be fine as is. The goal of the test is to check if the given `ext.provider` will be used to generate the EID. In the test we provide 

```
'ext': {'provider': 'some.provider.com'}
```

thus we expect the same provider in resulting EID.

File: test/spec/modules/eids_spec.js (hunk 1 above)
Comment (peixunzhang) [reply test/spec/modules/eids_spec.js]: Its js magic

File: test/spec/modules/eids_spec.js (hunk 2 above)
Comment (peixunzhang) [reply test/spec/modules/eids_spec.js:258]: Got it, thank you.

**Comment by 3link** [comment]
Opened a PR in Prebid: https://github.com/prebid/Prebid.js/pull/10783

**Pull request closed without merging**